### PR TITLE
feat(waylib): add sgrenderer damage tracking and vendored wlroots renderer

### PR DESCRIPTION
**Pull request by Groveer**
## Summary

为 waylib 实现全新的 QSGRenderer，支持 damage 跟踪并 vendor wlroots 渲染后端。

Implement a new QSGRenderer for waylib with damage tracking and vendored wlroots renderer backends.

### Phase 1 — RHI path damage tracking

- Add `WSGDamageTracker`: collects dirty nodes via `nodeChanged()` callback (Qt 6.8 removed `QSGNode::m_dirtyState`), accumulates bounding boxes with `DirtyDamageMask` including `DirtyMatrix`
- Add `WSGDirtyNodeObserver`: passive `QSGAbstractRenderer` registered on root node to receive `nodeChanged()` without displacing the real renderer
- Wire into `WBufferRenderer::ensureRenderer/render/destroySource`
- Replace `m_damageRing.add_whole()` with scanned dirty-node damage, combined with `rotate_buffer()` damage
- Add Y-flip for `mirrorVertically()` scenarios
- Add unit tests (18 cases covering all dirty flags, edge cases, Y-flip)

### Phase 2 — vendored wlroots 0.19 renderer

- Vendor gles2/vulkan/pixman backends, egl, dmabuf, allocator, swapchain infrastructure under `sgrenderer/vendor/wlroots/`
- Extension headers exposing buffer→FBO/Image conversion, texture handles, EGL context management for BAL consumption
- CMake target `waylib_sgrenderer_wlroots_vendor` (static lib)
- All Code Review fixes applied: underscore shader header names, `wlr_egl_context` for surface access, conditional `WLR_HAS_GBM_ALLOCATOR`, unified `config.h` includes, `glslangValidator FATAL_ERROR`

Both phases passed Code Review (18 unit tests).

Multica issue: WM-79

### CODE REVIEW

File: waylib/src/server/sgrenderer/vendor/wlroots/include/render/allocator/drm_dumb.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
+// SPDX-FileCopyrightText: 2017-2023 wlroots contributors
```
Comment (zzxyb) [inline waylib/src/server/sgrenderer/vendor/wlroots/include/render/allocator/drm_dumb.h:1]: allocator不需要copy,可以直接使用wlroots,或者自己扩展实现

File: waylib/src/server/sgrenderer/vendor/wlroots/render/wlr_renderer.c [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+#include <assert.h>
+#include <fcntl.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <wlr/backend.h>
+#include <wlr/render/interface.h>
+#include <wlr/render/pixman.h>
+#include <wlr/render/wlr_renderer.h>
+#include <wlr/types/wlr_drm.h>
```
Comment (zzxyb) [inline waylib/src/server/sgrenderer/vendor/wlroots/render/wlr_renderer.c:17]: 不应该实现wlr_renderer。只需要实现pixman\vulkan\gles后端

File: waylib/src/server/sgrenderer/vendor/wlroots/render/wlr_texture.c [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+#include <drm_fourcc.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+#include <wlr/render/interface.h>
+#include <wlr/render/wlr_texture.h>
+#include "render/pixel_format.h"
+#include "types/wlr_buffer.h"
+
+void wlr_texture_init(struct wlr_texture *texture, struct wlr_renderer *renderer,
```
Comment (zzxyb) [inline waylib/src/server/sgrenderer/vendor/wlroots/render/wlr_texture.c:18]: 同wlr_renderer一样只需要后端实现

**Review by zzxyb — CHANGES_REQUESTED** [review]
建议参考wlroots,然后以C++风格去写，这样可以避免引入wlroots很多功能结构和文件

**Comment by zzxyb** [comment]
pixman要评估一下，是否可以不用自己实现？用wlroot上的wlr_pixman_renderer_create
